src/main/tools/linux-sandbox-pid1.cc

```diff
@@ -87,6 +87,15 @@ static void CreateFile(const char *path) {
 // certain filesystems (e.g. XFS).
 static void LinkFile(const char *path) {
   if (link("tmp/empty_file", path) < 0) {
+    if (errno == EEXIST) {
+      // Try to remove the existing file first.
+      if (unlink(path) < 0) {
+        DIE("unlink %s", path);
+      }
+      if (link("tmp/empty_file", path) == 0) {
+        return;
+      }
+    }
     DIE("link %s", path);
   }
 }
```

Comment:
I'm assuming this is some performance critical code because it has a comment explaining to use link vs open for whatever small performance win that gives you.

If that is the case but maybe this option is still possible to write something that is a bit easier to run.
(Prepares state)

```
// ignore error for doesn't exist
unlink(path)
// must work now
link("tmp/empty_file", path")
```

Reply:
Isn't that slower though because it always runs two syscalls, even in the common case where the file doesn't exist yet? The fallback should be free if not taken.

Reply:
Right; I'm not sure how performance critical this code is.
I gather "it must be" since it's already trying to do link vs open but i read https://blogsystem5.substack.com/p/whatever-happened-to-sandboxfs recently and it made me think that performance of the sandbox is not well understood :p

(I always try to write dumb code first...)

Reply:
Thanks for confirming, then I do think this is right approach.